src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HttpHubConnectionBuilder.java

```diff
@@ -113,12 +115,25 @@ public HttpHubConnectionBuilder withHeader(String name, String value) {
         return this;
     }
 
+    /**
+     * Sets a method that will be called when constructing the HttpClient to allow customization such as certificate validation, proxies, and cookies.
+     * By default the client will have a cookie jar added and a read timeout for LongPolling.
+     *
+     * @param configureBuilder Callback for configuring the OkHttpClient.Builder.
+     * @return This instance of the HttpHubConnectionBuilder.
+     */
+    public HttpHubConnectionBuilder setConfigureHttpClient(Action1<OkHttpClient.Builder> configureBuilder) {
```

Comment:
I liked using `set` to make it clear this is effectively property setter, but seeing it paired with the verb "configure" is making me hesitant. Generally, I'd want to be setting a noun.

For example, I think `setHttpClientConfigurationCallback` sounds better despite maybe being a bit too wordy.

Does anyone have any other suggestions? Am I the only one who cares about the noun vs verb thing?

Reply:
> Am I the only one who cares about the noun vs verb thing?

You're not alone.

`setHttpClientBuilderCallback`?

Reply:
```suggestion
    public HttpHubConnectionBuilder setHttpClientBuilderCallback(Action1<OkHttpClient.Builder> configureBuilder) {
```